### PR TITLE
Allow model names with spaces to be a captured by the SetPropertyCommand on creation

### LINKED ISSUE #11227: Models mangles model names containing whitespace
There is a bug within the SetPropertyCommand class as described below:

>I'm having trouble using the command language to find a module with a space in its name. For example, the Wheat example has >a manager component that used to work like ``[Sow using a variable rule].Script.CultivarName = Batavia``, but the [regex to find >module names](https://github.com/APSIMInitiative/ApsimX/blob/4bbcea140ac54845dd55caddff2bfba209c51562/APSIM.Core/CommandProcessor/Language/SetPropertyCommandLanguage.cs#L20) mangles the component name at the last space.  
>
>Any clues appreciated.

_Originally posted by @peter-devoil in https://github.com/APSIMInitiative/ApsimX/discussions/10693#discussioncomment-16738801_

### PR DESCRIPTION
**Pull request by ric394**
resolves #11227

This change now allows the SetPropertyCommand to capture model names that include whitespace.

### CODE REVIEW

**Comment by par456** [comment]
Can I get a unit test for this edge case so we don't break it again later?

File: APSIM.Core/CommandProcessor/Language/SetPropertyCommandLanguage.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public partial class SetPropertyCommand: IModelCommand
     /// </remarks>
     public static IModelCommand Create(string command, string relativeToDirectory)
     {
-        string modelNameWithBrackets = @"[\w\d\[\]\.\:]+";
+        string modelNameWithBrackets = @"[\w\d\[\]\.\: ]*[\w\d\[\]\.\:][\w\d\[\]\.\: ]*";
 
         string pattern = $@"(?<keyword>{modelNameWithBrackets})\s*(?<operator>=|\+=|-=)\s*(?<pipe>\<)*\s*(?<value>[^\<]+)*";
```
[the remark this answers is not in the document]
Comment (ric394) [reply APSIM.Core/CommandProcessor/Language/SetPropertyCommandLanguage.cs]: Good suggestion! I'll implement this as suggested.
